Replace app icons with new, lighter set

snide opened the pull request:
cc to @gjones, @ryankeairns, @formgeist, @cchaos, @JessSmithSup, @Michalwo, @Adrian-J that this is incoming. I believe we'll have a solutions / product set following up in a bit as well.

### Summary

Switcheroo of EUI's app icon set based on @JessSmithSup's work. Process was the following.

1. Committed the new icons into Abstract, replacing our current set.
2. Used SVGO to compress them.
3. Removes the dark fill on the individual paths and removed fill="none" on the svg itself. This allows them to inherit the text color and make them "white" for dark mode.
4. Updated TS and usual odds and ends.
5. Applied a transform hack on the svg (figured this was better than `.euiIcon` itself, which could have its own animations applied) to fix some blurriness on Firefox. Hat tip to @cchaos for saving me a couple hours on that one 🎩 

#### Issues

* Icons are missing for grok debugger and multi-metric job. I think what is the population job now is likely supposed to be multi metric and possibly we're missing the population one?
* Softness on infra app because of the source file. I think I can fix this.

#### Questions

* @cchaos brought up some points about possibly reversing the colorscheme for visualize and watcher, but looking at them larger I _think_ they make sense as they are? It seems like we use the darker color for the outer elements. Dunno. could go either way.
* Should the various beats be named something unique? Possibly a beats prefix?

![image](https://user-images.githubusercontent.com/324519/46327398-52487480-c5b6-11e8-87d1-d98ac59f1548.png)

![image](https://user-images.githubusercontent.com/324519/46327401-56749200-c5b6-11e8-8bb6-f1358f5ab2ba.png)


### Checklist

- [x] This was checked in mobile
- [x] This was checked in IE11
- [x] This was checked in dark mode
- [ ] ~Any props added have proper autodocs~
- [x] Documentation examples were added
- [ ] A [changelog](https://github.com/elastic/eui/blob/master/CHANGELOG.md) entry exists and is marked appropriately
- [ ] This was checked for breaking changes and labeled appropriately
- [x] Jest tests were updated or added to match the most common scenarios
- [ ] ~This was checked against keyboard-only and screenreader scenarios~

formgeist commented:
@JessSmithSup New icons look spiffy 👍

JessSmithSup commented:
**Teal color change** - If we use these on the marketing site we aren’t going to change the color as we already have icons in this style and they could possibly live together. I prefer no shift, or if we are worried about it I’d rather just use a one color icon

**Reversing colors** - I’m totally ok with reversing the colors.

**Visualize icon** - If we feel strongly we can go back, my opinion is keeping these icons simple is best. We don’t have to make each icon a checklist of every functionality of the feature. The fact that this icon is distinct and also simple is good with me, adding more elements will make the level of detail inconsistent.

cchaos commented:
**Teal color change** - Yeah I'm not sure if it matters much, but thought I'd at least make people aware.

**Visualize icon** - I agree that keeping the icons simple is a good thing but I don't think my suggestion was any more complicated or noisy than most of the other icons (like index management). I do feel kind of strongly mainly because the vis app is such a main app and the icon is just kind of meh. (And no, it's not just because I had thought up the icon 6 months ago, I promise ;). I really do think it's a better representative of the entire vis app.

JessSmithSup commented:
@cchaos lmk if you're down with this

<img width="198" alt="screen shot 2018-10-02 at 2 06 28 pm" src="https://user-images.githubusercontent.com/21990848/46377050-78b8ef00-c64c-11e8-9d39-6d61b3d85034.png">

cchaos commented:
I'm digging it. Would it be simpler if the dots weren't hollow?

JessSmithSup commented:
@cchaos that's how this set is treating circles so I'd like to keep that part consistent

cchaos commented:
@JessSmithSup Thoughts then on changing the hollow bars to lines like the ones representing lines on paper and the canvas app?

JessSmithSup commented:
@cchaos I think I'm into it...
<img width="268" alt="screen shot 2018-10-02 at 3 49 05 pm" src="https://user-images.githubusercontent.com/21990848/46381372-d05e5700-c65a-11e8-8fee-e7416723633f.png">

snide commented:
OK. Latest update made the following changes.

1. Added new grok and multi-metric icons.
2. Adjust teal fills to use the EUI palette (the brand teal is too hot against our blue).
3. Use classes to define the teal coloring on the individual SVG paths. This lets us overwrite the coloring by applying the "text" color to make them mono in the few cases it makes sense to use them that way.

**Ignore timelion icon being old. It's a Kibana thing**

## After

![image](https://user-images.githubusercontent.com/324519/46381305-968d5080-c65a-11e8-8677-4693b718352d.png)

## Before

![image](https://user-images.githubusercontent.com/324519/46381324-a86ef380-c65a-11e8-809d-3e01a6a7f5b9.png)

## Color overwrite documentation

![image](https://user-images.githubusercontent.com/324519/46381363-c76d8580-c65a-11e8-8d97-84feadfae6f1.png)

cchaos commented:
@JessSmithSup Too many lines?

<img width="155" alt="screen shot 2018-10-02 at 18 53 04 pm" src="https://user-images.githubusercontent.com/549577/46381506-969a4a00-c674-11e8-913d-6ce523f91bab.png">

snide commented:
jenkins, test this

JessSmithSup commented:
@cchaos yeah, I think maybe keeping it at 3 is nice.

@snide breaking my heart with that color change lol. Are you saying the teal is too hot next to the blue labels? Shifting this goes away from the consistency of branding. Any reason those labels aren't bold black like the ones on top? I guess because they are links? Not sure I'm anymore into that dark teal next to the dark blue. I think a lighter teal is a nicer contrast. I would lobby against a dark shift.

snide commented:
Lemme spin up some tests. I'll get you some screens so you can see it a couple different ways.

> Any reason those labels aren't bold black like the ones on top?

Yep. That's the reason.

ryankeairns commented:
My personal opinion is that the EUI primary blue and teal in the icons clash - they simply don't work well together. Short term, I favor the mono icon; longer term, a page redesign should be considered as there is a lot going on here with the number, size, and proximity of icons to one another.

cchaos commented:
As Ryan said, the main issue is that the new teal clashes with our primary blue which is different from the marketing primary blue which is much lighter.

<img width="193" alt="screen shot 2018-10-03 at 10 21 11 am" src="https://user-images.githubusercontent.com/549577/46416711-13213d00-c6f6-11e8-9c2c-df7d82893c32.png">

We can't lighten our primary blue to match because our buttons and text must be AA contrast level compliant. That makes the teal clash really with our whole color scheme because of we've got such darker muted colors for accessibility. I don't think it's a big deal to darken the teal for use in Kibana as we don't (and can't) match the marketing site anyway.

We might be able to get away with something in between our secondary color and this teal. I mentioned that `#00A69B` darkened the teal just enough so that it was  AA 18+ compliant. Or try desaturating the teal by 50%.

The other option is just to swap the teal for primary blue?

JessSmithSup commented:
I think I'm maybe not feeling any of the teal/blue mixes. Monochromatic would be cleaner. Controversial thought for this page.... what if they were monochromatic EUI primary blue?  Either that or keep it all monochromatic dark cool gray

JessSmithSup commented:
Also we're shifting our brand main blue #0080d5 and the darker swatches are #006cb8 and #00629e. And on the brand side we haven't explored how to handle these icons next to a blue link, so, maybe it is something that shifts to blue when it is associated with a link on the brand side? Sorry for all of these long-winded responses :)

Michalwo commented:
> Also we're shifting our brand main blue #0080d5 and the darker swatches are #006cb8 and #00629e. And on the brand side we haven't explored how to handle these icons next to a blue link, so, maybe it is something that shifts to blue when it is associated with a link on the brand side? Sorry for all of these long-winded responses :)

@JessSmithSup ^^ good point and worth exploring.

snide commented:
@cchaos correct.

### Mono
![image](https://user-images.githubusercontent.com/324519/46427714-f6393980-c6f6-11e8-9e70-e54b1a4a49eb.png)

### Primary
![image](https://user-images.githubusercontent.com/324519/46427708-f1748580-c6f6-11e8-8fc8-f164ef8b9076.png)

### Secondary (which will be the default)
![image](https://user-images.githubusercontent.com/324519/46427727-fa655700-c6f6-11e8-86c8-629ec9c9a262.png)

cchaos commented on src-docs/src/views/icon/apps.js:
```diff
@@ -37,6 +37,18 @@ const iconTypes = [
+  'notebookApp',
+  'packetbeatApp',
+  'securityAnalyticsApp',
+  'sqlApp',
```
I would still love to these alphabetized. VSCode has a really easy way to do it: `command + shift + p -> sort lines`

cchaos commented on src-docs/src/views/icon/icon_colors.js:
```diff
@@ -43,5 +43,20 @@ export default () => (
       ))
     }
+    <EuiFlexItem
+      className="guideDemo__icon"
```
I know this was a quick add but the layout of this one bugs me, I'd really rather just see it in it's own row/grid.

<img width="980" alt="screen shot 2018-10-03 at 13 37 15 pm" src="https://user-images.githubusercontent.com/549577/46428437-f7c42b00-c711-11e8-9571-5709631495e1.png">

cchaos commented on CHANGELOG.md:
```diff
@@ -1,12 +1,13 @@
 - Added TypeScript typings for `EuiKeyPadMenu` ([#1229](https://github.com/elastic/eui/pull/1229))
-- Force `EuiPopover` contents to stick to its initial position when the content changes ([#1199](https://github.com/elastic/eui/pull/1199))
+- Forced `EuiPopover` contents to stick to its initial position when the content changes ([#1199](https://github.com/elastic/eui/pull/1199))
+- Updated `EuiIcon` app icon set and allow them to adjust two-toned colorschemes ([#1225](https://github.com/elastic/eui/pull/1225))
```
No longer adjusting the "two-toned" color [ ] schemes.

snide commented:
For those playing at home, here's how things finally landed with the coloring of the icons. @cchaos this also addresses your feedback on docs.

![image](https://user-images.githubusercontent.com/324519/46430442-cb9eaf00-c6fd-11e8-9b5a-35a703165fd0.png)

JessSmithSup commented:
Apologies for the latest response ever. It was more difficult than I thought to get stable internet in an airport. This all looks/sounds good. Thanks for putting so much thought into it!
